### PR TITLE
VSP-443 [iOS] My Archives - Pending archives

### PR DESCRIPTION
**Pull request by luciancerbu-vsp**
Added UI elements for pending archives.
Added API requests functionality for pending archives (accept/decline buttons).
Small UI changes to pending archives tableview section.

### CODE REVIEW

File: Permanent/ViewControllers/ArchivesViewController.swift [hunk 1]
```diff
@@ -219,67 +219,137 @@ class ArchivesViewController: BaseViewController<ArchivesViewModel> {
...
-                                tableView.reloadData()
-                            } else {
-                                self?.showAlert(title: .error, message: .errorMessage)
-                            }
+        if tableViewSections[indexPath.section] == .ok {
+            if let tableViewCell = tableView.dequeueReusableCell(withIdentifier: String(describing: ArchiveScreenChooseArchiveDetailsTableViewCell.self)) as? ArchiveScreenChooseArchiveDetailsTableViewCell,
+               let tableViewData = viewModel?.selectableArchives {
+                let archiveVO = tableViewData[indexPath.row]
+                tableViewCell.updateCell(withArchiveVO: archiveVO, isDefault: archiveVO.archiveID == viewModel?.defaultArchiveId, isManaging: isManaging)
+                tableViewCell.rightButtonAction = { [weak self] cell in
```
Comment (v-rusu) [inline Permanent/ViewControllers/ArchivesViewController.swift:241]: Let's move the button actions to separate functions. This one is getting a little bit too cluttered

Comment (v-rusu) [reply Permanent/ViewControllers/ArchivesViewController.swift:241]: check the button actions below as well

File: Permanent/ViewControllers/ArchivesViewController.swift [hunk 2]
```diff
@@ -219,67 +219,137 @@ class ArchivesViewController: BaseViewController<ArchivesViewModel> {
...
+        }
+        if let selectableArchives = viewModel?.selectableArchives,
+           selectableArchives.count > 0{
+            tableViewSections.append(ArchiveVOData.Status.ok)
+        }
+        return tableViewSections.count
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+
```
Comment (v-rusu) [inline Permanent/ViewControllers/ArchivesViewController.swift:346]: please remove this white line

File: Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenPendingArchiveDetailsTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        
+        archiveNameLabel.font = Text.style16.font
+        archiveNameLabel.textColor = .darkBlue
+        
+        archiveAccessLabel.font = Text.style8.font
+        archiveAccessLabel.textColor = .darkBlue
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        contentView.backgroundColor = .white
```
Comment (v-rusu) [inline Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenPendingArchiveDetailsTableViewCell.swift:37]: Selection of a row should be blocked in the corresponding tableView methods. shouldSelectRowAt: and shouldHighlightRowAt:

File: Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenTopSectionTitleTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+//  ArchiveScreenTopSectionTableViewCell.swift
+//  Permanent
+//
+//  Created by Lucian Cerbu on 15.09.2021.
+//  Copyright © 2021 Victory Square Partners. All rights reserved.
+//
+
+import UIKit
+
+class ArchiveScreenSectionTitleTableViewCell: UITableViewCell {
```
Comment (v-rusu) [inline Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenTopSectionTitleTableViewCell.swift:11]: This looks like a duplicate definition of ArchiveScreenSectionTitleTableViewCell, but with a different filename

File: Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenSectionTitleTableViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,40 @@
+//
+//  ArchiveScreenTopSectionTableViewCell.swift
+//  Permanent
+//
+//  Created by Lucian Cerbu on 15.09.2021.
+//
+
+import UIKit
+
+class ArchiveScreenSectionTitleTableViewCell: UITableViewCell {
```
Comment (v-rusu) [inline Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenSectionTitleTableViewCell.swift:10]: this should be a subclass of UITableViewHeaderFooterView

Comment (luciancerbu-vsp) [reply Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenSectionTitleTableViewCell.swift:10]: Changed UITableViewCell with UITableViewHeaderFooterView.

File: Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenPendingArchiveDetailsTableViewCell.swift (hunk 3 above)
Comment (luciancerbu-vsp) [reply Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenPendingArchiveDetailsTableViewCell.swift:37]: Removed `setSelected` method from ArchiveScreenPendingArchiveDetailsTableViewCell.swift and added `tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool ` in ArchivesViewController.swift

Where shouldSelectRowAt be used?

File: Permanent/ViewControllers/ArchivesViewController.swift (hunk 2 above)
Comment (luciancerbu-vsp) [reply Permanent/ViewControllers/ArchivesViewController.swift:346]: White line was removed.

File: Permanent/ViewControllers/ArchivesViewController.swift (hunk 1 above)
Comment (luciancerbu-vsp) [reply Permanent/ViewControllers/ArchivesViewController.swift:241]: Created `rightButtonAction() acceptButtonAction() declineButtonAction()` methods for tableview buttons.

File: Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenTopSectionTitleTableViewCell.swift (hunk 4 above)
Comment (luciancerbu-vsp) [reply Permanent/Views/Cells/ArchiveTableViewCell/ArchiveScreenTopSectionTitleTableViewCell.swift]: ArchiveScreenTopSectionTitleTableViewCell.swift was deleted.

**Pull request merged**